### ISSUE TITLE
Request to make ngx-build-plus compatible with Angular CLI V 13

### ISSUE THREAD

**Issue by naveedahmed1** [manfredsteyer/ngx-build-plus#307]
I recently upgraded my Angular app to the latest version.

After upgrade when I try to build my app it throws below error:

```
An unhandled exception occurred: Cannot destructure property 'styles' of 'buildOptions.sourceMap' as it is undefined.
See "C:\Users\myuser\AppData\Local\Temp\ng-AkN2KG\angular-errors.log" for further details.

[error] TypeError: Cannot destructure property 'styles' of 'buildOptions.sourceMap' as it is undefined.
at getBrowserConfig (D:\MyProject\node_modules@angular-devkit\build-angular\src\webpack\configs\browser.js:18:21)
at D:\MyProject\node_modules@angular-devkit\build-angular\src\builders\dev-server\index.js:127:44
at D:\MyProject\node_modules@angular-devkit\build-angular\src\utils\webpack-browser-config.js:71:16
at generateWebpackConfig (D:\MyProject\node_modules@angular-devkit\build-angular\src\utils\webpack-browser-config.js:60:40)
at async generateBrowserWebpackConfigFromContext (D:\MyProject\node_modules@angular-devkit\build-angular\src\utils\webpack-browser-config.js:122:20)
at async generateI18nBrowserWebpackConfigFromContext (D:\MyProject\node_modules@angular-devkit\build-angular\src\utils\webpack-browser-config.js:69:20)
at async setup (D:\MyProject\node_modules@angular-devkit\build-angular\src\builders\dev-server\index.js:124:47)
```

I opened an issue with Angular CLI https://github.com/angular/angular-cli/issues/21943 and they said:

> This seems to be an issue with ngx-build-plus which is not compatible with version 13 of the Angular CLI.
(https://github.com/angular/angular-cli/issues/21943#issuecomment-942956408)

I hope this can be fixed in next version of `ngx-build-plus`

Thank you!

**Comment by AleksanderBodurri** [comment]
@Timebutt  I've encountered this issue as well, which is why I opened https://github.com/manfredsteyer/ngx-build-plus/pull/305.

**Comment by Timebutt** [comment]
Hey @AleksanderBodurri, I see. I just compared our PRs and see that you should also remove `hmrWarning` and `servePathDefaultWarning` as far as I know. I removed them in my PR, yours still have those options. Ideally we should consolidate our efforts and get either PR merged, what's your preference?

**Comment by AleksanderBodurri** [comment]
It'll be easier to go with your PR. I'll close mine and leave a closing comment referencing yours @Timebutt

**Comment by samartomar** [comment]
In urgent need for this change or workaround without downgrading angular version.

**Comment by Timebutt** [comment]
@samartomar have a look at my PR, if you manually apply it to the file in `node_modules` and then use [`patch-package`](https://www.npmjs.com/package/patch-package) you have a fix until `ngx-build-plus` is released.

**Comment by guisperandio** [comment]
Any update on this @manfredsteyer ?

**Comment by pablomario** [comment]
Hi @naveedahmed1 !

I have found a possible solution (At least it is a valid solution for me).
The solution is simple, create a new project that has the same architecture configuration as your project using 'ng new you-app-name'.
Now overwrite the new 'angular.json' file in your project (You may have to change some testing or configuration path)
Run 'ng serve' again and it should work.

**Comment by naveedahmed1** [comment]
This issue has been fixed in V 13.0.1.
